pkg/scheduler/internal/queue/scheduling_queue_test.go

```diff
@@ -323,15 +324,47 @@ func TestPriorityQueue_Update(t *testing.T) {
 	if p, err := q.Pop(); err != nil || p.Pod != highPriNominatedPodInfo.Pod {
 		t.Errorf("Expected: %v after Pop, but got: %v", highPriorityPodInfo.Pod.Name, p.Pod.Name)
 	}
-	// Updating a pod that is in unschedulableQ in a way that it may
-	// become schedulable should add the pod to the activeQ.
+
+	// Updating a pod that is in backoff queue and it is still backing off
+	// pod will not be moved to active queue, and it will be updated in backoff queue
+	podInfo := q.newQueuedPodInfo(medPriorityPodInfo.Pod)
+	if err := q.podBackoffQ.Add(podInfo); err != nil {
+		t.Errorf("adding pod to backoff queue error: %v", err)
+	}
+	q.Update(podInfo.Pod, podInfo.Pod)
+	rawPodInfo, err := q.podBackoffQ.Pop()
+	podGotFromBackoffQ := rawPodInfo.(*framework.QueuedPodInfo).Pod
+	if err != nil || podGotFromBackoffQ != medPriorityPodInfo.Pod {
+		t.Errorf("Expected: %v after Pop, but got: %v", medPriorityPodInfo.Pod.Name, podGotFromBackoffQ.Name)
+	}
+
+	// updating a pod which is in unschedulable queue, and it is not backing off,
+	// we will move it to active queue
 	q.AddUnschedulableIfNotPresent(q.newQueuedPodInfo(medPriorityPodInfo.Pod), q.SchedulingCycle())
 	if len(q.unschedulableQ.podInfoMap) != 1 {
 		t.Error("Expected unschedulableQ to be 1.")
 	}
 	updatedPod := medPriorityPodInfo.Pod.DeepCopy()
 	updatedPod.ClusterName = "test"
 	q.Update(medPriorityPodInfo.Pod, updatedPod)
+	rawPodInfo, err = q.podBackoffQ.Pop()
```

Comment:
Shoulnd't we pop from activeQ in this cycle?

Reply:
sorry for the misleading comments, updated

Reply:
LGTM now.